test/CMakeLists.txt

```diff
@@ -150,6 +150,10 @@ if (pybind11_VERSION VERSION_LESS 2.5.99)
  list(REMOVE_ITEM binder_tests T11.override)
  list(REMOVE_ITEM binder_tests T15.inner_class)
 endif()
+if (LLVM_VERSION_MAJOR VERSION_GREATER 15)
+ list(REMOVE_ITEM binder_tests T15.inner_class)
+ list(REMOVE_ITEM binder_tests T17.anonymous)
```

Comment:
@andriish  why do we need to remove these?

Reply:
Those tests have issues with the newest compilers supplied by Fedora. I would suggest to merge it as is and later try to revert exactly this change to debug the issue.